From 79fa6495c4485cc95b993268fc16a217f759ebc7 Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Fri, 15 Apr 2016 08:26:32 -0400
Subject: [PATCH 26/33] arm: move aes_glue.h to the arch/arm/include/asm
 directory

The aes_glue.h defines function which might be used from drivers.
This commit moves the file to the arch/arm/include/asm to be accessible
publicly.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
[ j-stiffler: fixed include order of headers ]
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/crypto/aes_glue.c      |  3 +--
 arch/arm/crypto/aes_glue.h      | 19 -------------------
 arch/arm/crypto/aesbs-glue.c    |  3 +--
 arch/arm/include/asm/aes_glue.h | 19 +++++++++++++++++++
 4 files changed, 21 insertions(+), 23 deletions(-)
 delete mode 100644 arch/arm/crypto/aes_glue.h
 create mode 100644 arch/arm/include/asm/aes_glue.h

diff --git a/arch/arm/crypto/aes_glue.c b/arch/arm/crypto/aes_glue.c
index 0409b8f..94cab03 100644
--- a/arch/arm/crypto/aes_glue.c
+++ b/arch/arm/crypto/aes_glue.c
@@ -5,8 +5,7 @@
 #include <linux/module.h>
 #include <linux/crypto.h>
 #include <crypto/aes.h>
-
-#include "aes_glue.h"
+#include <asm/aes_glue.h>
 
 EXPORT_SYMBOL(AES_encrypt);
 EXPORT_SYMBOL(AES_decrypt);
diff --git a/arch/arm/crypto/aes_glue.h b/arch/arm/crypto/aes_glue.h
deleted file mode 100644
index cca3e51..0000000
--- a/arch/arm/crypto/aes_glue.h
+++ /dev/null
@@ -1,19 +0,0 @@
-
-#define AES_MAXNR 14
-
-struct AES_KEY {
-	unsigned int rd_key[4 * (AES_MAXNR + 1)];
-	int rounds;
-};
-
-struct AES_CTX {
-	struct AES_KEY enc_key;
-	struct AES_KEY dec_key;
-};
-
-asmlinkage void AES_encrypt(const u8 *in, u8 *out, struct AES_KEY *ctx);
-asmlinkage void AES_decrypt(const u8 *in, u8 *out, struct AES_KEY *ctx);
-asmlinkage int private_AES_set_decrypt_key(const unsigned char *userKey,
-					   const int bits, struct AES_KEY *key);
-asmlinkage int private_AES_set_encrypt_key(const unsigned char *userKey,
-					   const int bits, struct AES_KEY *key);
diff --git a/arch/arm/crypto/aesbs-glue.c b/arch/arm/crypto/aesbs-glue.c
index 6d68529..d84bdce 100644
--- a/arch/arm/crypto/aesbs-glue.c
+++ b/arch/arm/crypto/aesbs-glue.c
@@ -13,8 +13,7 @@
 #include <crypto/ablk_helper.h>
 #include <crypto/algapi.h>
 #include <linux/module.h>
-
-#include "aes_glue.h"
+#include <asm/aes_glue.h>
 
 #define BIT_SLICED_KEY_MAXSIZE	(128 * (AES_MAXNR - 1) + 2 * AES_BLOCK_SIZE)
 
diff --git a/arch/arm/include/asm/aes_glue.h b/arch/arm/include/asm/aes_glue.h
new file mode 100644
index 0000000..cca3e51
--- /dev/null
+++ b/arch/arm/include/asm/aes_glue.h
@@ -0,0 +1,19 @@
+
+#define AES_MAXNR 14
+
+struct AES_KEY {
+	unsigned int rd_key[4 * (AES_MAXNR + 1)];
+	int rounds;
+};
+
+struct AES_CTX {
+	struct AES_KEY enc_key;
+	struct AES_KEY dec_key;
+};
+
+asmlinkage void AES_encrypt(const u8 *in, u8 *out, struct AES_KEY *ctx);
+asmlinkage void AES_decrypt(const u8 *in, u8 *out, struct AES_KEY *ctx);
+asmlinkage int private_AES_set_decrypt_key(const unsigned char *userKey,
+					   const int bits, struct AES_KEY *key);
+asmlinkage int private_AES_set_encrypt_key(const unsigned char *userKey,
+					   const int bits, struct AES_KEY *key);
-- 
1.9.1

